recipes_source/android_native_app_with_custom_op.rst

```diff
@@ -0,0 +1,732 @@
+
+Building the app
+================
+
+To specify to gradle where is Android SDK and Android NDK, we need to fill ``NativeApp/local.properties``.
+
+.. code-block:: shell
+
+  cd NativeApp
+  echo "sdk.dir=$ANDROID_HOME" >> NativeApp/local.properties
+  echo "ndk.dir=$ANDROID_NDK" >> NativeApp/local.properties
+
+
+To build the result ``apk`` file we run:
+
+.. code-block:: shell
+
+  cd NativeApp
+  gradle app:assembleDebug
+
+To install the app on the connected device:
+
+.. code-block:: shell
+
+  cd NativeApp
+  gradle app::installDebug
+
+After that, you can run the app on the device by clicking on PyTorchNativeApp icon.
+Or you can do it from the command line:
+
+.. code-block:: shell
+
+  adb shell am start -n org.pytorch.nativeapp/.MainActivity
+
+If you check the android logcat:
+
+.. code-block:: shell
+
+  adb logcat -v brief | grep PyTorchNativeApp
+
+
+You should see logs with tag 'PyTorchNativeApp' that prints x, y, and the result of the model forward, which we print with ``log`` function in ``NativeApp/app/src/main/cpp/pytorch_nativeapp.cpp``.
+
+.. code-block::
+
+  I/PyTorchNativeApp(26968): x: -0.9484 -1.1757 -0.5832  0.9144  0.8867  1.0933 -0.4004 -0.3389
+  I/PyTorchNativeApp(26968): -1.0343  1.5200 -0.7625 -1.5724 -1.2073  0.4613  0.2730 -0.6789
+  I/PyTorchNativeApp(26968): -0.2247 -1.2790  1.0067 -0.9266  0.6034 -0.1941  0.7021 -1.5368
+  I/PyTorchNativeApp(26968): -0.3803 -0.0188  0.2021 -0.7412 -0.2257  0.5044  0.6592  0.0826
+  I/PyTorchNativeApp(26968): [ CPUFloatType{4,8} ]
+  I/PyTorchNativeApp(26968): y: -1.0084  1.8733  0.5435  0.1087 -1.1066
+  I/PyTorchNativeApp(26968): -1.9926  1.1047  0.5311 -0.4944  1.9178
+  I/PyTorchNativeApp(26968): -1.5451  0.8867  1.0473 -1.7571  0.3909
+  I/PyTorchNativeApp(26968):  0.4039  0.5085 -0.2776  0.4080  0.9203
+  I/PyTorchNativeApp(26968):  0.3655  1.4395 -1.4467 -0.9837  0.3335
+  I/PyTorchNativeApp(26968): -0.0445  0.8039 -0.2512 -1.3122  0.6543
+  I/PyTorchNativeApp(26968): -1.5819  0.0525  1.5680 -0.6442 -1.3090
+  I/PyTorchNativeApp(26968): -1.6197 -0.0773 -0.5967 -0.1105 -0.3122
+  I/PyTorchNativeApp(26968): [ CPUFloatType{8,5} ]
+  I/PyTorchNativeApp(26968): result:  16.0274   9.0330   6.0124   9.8644  11.0493
+  I/PyTorchNativeApp(26968):   8.7633   6.9657  12.3469  10.3159  12.0683
+  I/PyTorchNativeApp(26968):  12.4529   9.4559  11.7038   7.8396   6.9716
+  I/PyTorchNativeApp(26968):   8.5279   9.1780  11.3849   8.4368   9.1480
+  I/PyTorchNativeApp(26968):  10.0000  10.0000  10.0000  10.0000  10.0000
+  I/PyTorchNativeApp(26968):  10.0000  10.0000  10.0000  10.0000  10.0000
+  I/PyTorchNativeApp(26968):  10.0000  10.0000  10.0000  10.0000  10.0000
+  I/PyTorchNativeApp(26968):  10.0000  10.0000  10.0000  10.0000  10.0000
+  I/PyTorchNativeApp(26968): [ CPUFloatType{8,5} ]
+
+
+
+The full setup of this app you can find in `PyTorch Android Demo Application Repository <https://github.com/pytorch/android-demo-app/tree/master/NativeApp>`_.
```

Comment:
Recommend to also add this line at the beginning of the recipe for users to download and follow along from the start.